homeassistant/components/zeroconf/__init__.py

```diff
@@ -259,44 +275,95 @@ async def _async_register_hass_zc_service(
         )
 
 
-async def _async_start_zeroconf_browser(
-    hass: HomeAssistant, zeroconf: Zeroconf
-) -> None:
-    """Start the zeroconf browser."""
+class FlowDispatcher:
+    """Dispatch discovery flows."""
 
-    zeroconf_types = await async_get_zeroconf(hass)
-    homekit_models = await async_get_homekit(hass)
+    def __init__(self, hass: HomeAssistant):
+        """Init the discovery dispatcher."""
+        self.hass = hass
+        self.pending_flows: list[ZeroconfFlow] = []
+        self.started = False
 
-    types = list(zeroconf_types)
+    @callback
+    def async_start(self) -> None:
+        """Start processing pending flows."""
+        self.started = True
+        for flow in list(self.pending_flows):
+            self.hass.async_create_task(self._init_flow(flow))
+        self.pending_flows = []
 
-    for hk_type in HOMEKIT_TYPES:
-        if hk_type not in zeroconf_types:
-            types.append(hk_type)
+    def create(self, flow: ZeroconfFlow) -> None:
+        """Create and add or queue a flow."""
+        if self.started:
+            self.hass.create_task(self._init_flow(flow))
+        else:
+            self.pending_flows.append(flow)
+
+    def _init_flow(self, flow: ZeroconfFlow) -> Coroutine[Any, Any, Any]:
```

Comment:
Let's try to avoid `Any`.
```suggestion
    def _init_flow(self, flow: ZeroconfFlow) -> Coroutine[None, None, FlowResult]:
```

Reply:
Thanks